### PR TITLE
hash getRandomBytes before using as Seed for ChaChaDRG, closes #22

### LINKED ISSUE #22: hash entropy used in seedNew
The motivation is to mitigate any compromise of system entropy source (that can be not even an entropy but some substitute, in case DoNotUseEntropy flag is used).

This line: https://github.com/kazu-yamamoto/crypton/blob/master/Crypto/Random.hs#L52

Replace this:

```
seedNew = Seed `fmap` getRandomBytes seedLength
```

with something like:

```
seedNew = Seed . take seedLength . BA.convert . (hash :: ByteString -> Digest SHA512) `fmap` getRandomBytes 64
```

It assumes that seedLength is less than 64 bytes, it's 40 in this file, but we could of course use some more "clever" approach and define something like getHashedRandomBytes that would get "random" bytes in 32/64 byte chunks, hash (256/512) and append them, and throw away what was unused. Given that more than 64 bytes almost never needed, it seems overkill.

cc @kazu-yamamoto - that would certainly be a good incentive to switch from cryptonite :)))

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: Crypto/Random.hs [hunk 1]
```diff
@@ -49,7 +50,7 @@ seedLength = 40
 
 -- | Create a new Seed from system entropy
 seedNew :: MonadRandom randomly => randomly Seed
-seedNew = Seed `fmap` getRandomBytes seedLength
+seedNew = (Seed . B.take seedLength . B.convert . (hash :: ScrubbedBytes -> Digest SHA512)) `fmap` getRandomBytes 64
```
Comment (kazu-yamamoto) [inline Crypto/Random.hs:53]: As I said, please write a comment why hashing is necessary here.

Comment (intricate) [reply Crypto/Random.hs:53]: @epoberezkin would you be willing to do this? If not, I can do so. I'd essentially just copypaste your argument from #22

**Comment by kazu-yamamoto** [comment]
Added a comment and merged.
Thanks!

**Comment by epoberezkin** [comment]
thank you, sorry - missed the question!

**Pull request merged**